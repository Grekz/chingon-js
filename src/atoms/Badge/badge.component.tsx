import styled from '@emotion/styled'
import * as React from 'react'

interface IBadgeProps {
  status: BadgeStatusType
  content?: string
  fontColor?: string
  size?: string
}

export type BadgeStatusType = 'online' | 'away' | 'busy' | 'offline'

const Badge = ({ status, content, fontColor, size }: IBadgeProps) => {
  return (
    <BadgeWrapper status={status} fontColor={fontColor} size={size}>
      {content ? <span>{content}</span> : null}
    </BadgeWrapper>
  )
}

Badge.defaultProps = {
  fontColor: '#fff',
  size: '30px',
}

const getStatusColor = (status: BadgeStatusType) => {
  switch (status) {
    case 'online':
      return 'green'
    case 'away':
      return 'yellow'
    case 'busy':
      return 'red'
    case 'offline':
      return 'lightgray'
    default:
      return 'lightgray'
  }
}

const BadgeWrapper = styled.div`
  background-color: ${({ status }: IBadgeProps) => getStatusColor(status)};
  padding: 0.5rem;
  border-radius: 50%;
  color: ${({ fontColor }: IBadgeProps) => fontColor};
  width: ${({ size }: IBadgeProps) => size};
  height: ${({ size }: IBadgeProps) => size};
  display: inline-flex;
  align-content: center;
  text-align: center;
  flex-flow: row wrap;
  vertical-align: top;
  > span {
    width: 100%;
  }
`

export default Badge
