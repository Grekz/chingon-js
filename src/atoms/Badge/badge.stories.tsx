import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Badge from '.'

storiesOf('Atoms|Badge', module).add('As Default', () => (
  <Badge status="online" content="4" />
))
