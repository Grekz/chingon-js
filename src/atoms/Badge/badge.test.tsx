import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Badge from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Badge /> rendering', () => {
  it('should render Badge', () => {
    const wrapper = shallow(<Badge status="online" content="4" />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<Badge status="online" content="4" />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
