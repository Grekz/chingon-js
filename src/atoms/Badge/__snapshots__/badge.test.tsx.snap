// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> rendering should match snapshot 1`] = `
<Styled(div)
  fontColor="#fff"
  size="30px"
  status="online"
>
  <span>
    4
  </span>
</Styled(div)>
`;
